platforms/documentation/docs/src/docs/userguide/optimizing-builds/performance.adoc

```diff
@@ -15,484 +15,622 @@
+WARNING: Gradle’s Groovy DSL relies on Groovy’s dynamic features. To use static compilation in plugins, adopt a more Java-like syntax.
 
-The following example defines a task that copies files without dynamic features:
+The example below defines a task that copies files without dynamic features:
 
-====
-.src/main/groovy/MyPlugin.groovy
 [source,groovy]
+.src/main/groovy/MyPlugin.groovy
 ----
 project.tasks.register('copyFiles', Copy) { Task t ->
     t.into(project.layout.buildDirectory.dir('output'))
     t.from(project.configurations.getByName('compile'))
 }
 ----
-====
 
-This example uses the `register()` and `getByName()` methods available on all Gradle "domain object containers".
-Domain object containers include tasks, configurations, dependencies, extensions, and more.
-Some collections, such as `TaskContainer`, have dedicated types with extra methods like link:{groovyDslPath}/org.gradle.api.tasks.TaskContainer.html#org.gradle.api.tasks.TaskContainer:create(java.lang.String,%20java.lang.Class)[create],
-which accepts a task type.
+This example uses `register()` and `getByName()`, available on all Gradle *domain object containers*, such as tasks, configurations, dependencies, and extensions.
+Some containers, like `TaskContainer`, have specialized methods such as link:{groovyDslPath}/org.gradle.api.tasks.TaskContainer.html#org.gradle.api.tasks.TaskContainer:create(java.lang.String,%20java.lang.Class)[create], which accepts a task type.
 
-When you use static compilation, an IDE can:
+Using static compilation improves IDE support by enabling:
 
-* quickly show errors related to unrecognised types, properties, and methods
-* auto-complete method names
+- Faster detection of unrecognized types, properties, and methods
+- More reliable auto-completion for method names
 
 [[optimize_dependency_resolution]]
-== Optimize Dependency resolution
+== 10. Optimize dependency resolution
+
+Dependency resolution simplifies integrating third-party libraries into your projects.
+Gradle contacts remote servers to discover and download dependencies.
+You can optimize how dependencies are referenced to minimize these remote calls.
 
-Dependency resolution simplifies integrating third-party libraries and other dependencies into your projects.
-Gradle contacts remote servers to discover and download dependencies. You can optimize the way you reference
-dependencies to cut down on these remote server calls.
+=== Avoid Unnecessary and Unused Dependencies
 
-=== Avoid unnecessary and unused dependencies
+Managing third-party libraries and their transitive dependencies adds significant maintenance and build time costs.
 
-Managing third-party libraries and their transitive dependencies adds a significant
-cost to project maintenance and build times.
+Unused dependencies often remain after refactors.
+To identify and remove them, use the https://github.com/nebula-plugins/gradle-lint-plugin[Gradle Lint plugin].
```

Comment:
While this is being updated, it might be good to mention that this plugin doesn't support the Gradle Kotlin DSL - https://github.com/nebula-plugins/gradle-lint-plugin/issues/166

Reply:
Removed entirely.